### PR TITLE
imagemagick: 6.9.11-29 -> 6.9.11-57

### PR DESCRIPTION
**Pull request by ajs124**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
See #109184 and #100302.
According to https://nvd.nist.gov/vuln/detail/CVE-2020-29599, this fixes at least CVE-2020-29599.

File: pkgs/applications/graphics/ImageMagick/default.nix [hunk 1]
```diff
@@ -12,39 +12,19 @@ let
...
-    version = "6.9.11-29";
-    sha256 = "0adjdpi91ya0g3v7y503n95833m25aibkim5swg5lnwjrba407hg";
-    patches = [];
-  }
-    # Freeze version on mingw so we don't need to port the patch too often.
-    # FIXME: This version has multiple security vulnerabilities
-    // lib.optionalAttrs (stdenv.hostPlatform.isMinGW) {
-        version = "6.9.2-0";
-        sha256 = "17ir8bw1j7g7srqmsz3rx780sgnc21zfn0kwyj78iazrywldx8h7";
-        patches = [(fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/ImageMagick/default.nix:26]: Please remove fetchpatch from inputs.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109244` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>imagemagick</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

imagemagick:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Magick++-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Magick++-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Magick-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Magick-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/MagickCore-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/MagickCore-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/MagickWand-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/MagickWand-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Wand-config'
substituteStream(): WARNING: pattern '/nix/store/8d70a1dc03f1j4lvddf66faahnrg4bzy-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/mcxksaa14i9m2d16z80dp87b4mz10nrg-imagemagick-6.9.11-57-dev/bin/Wand-config'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109244` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>imagemagick</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

imagemagick:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Magick++-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Magick++-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Magick-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Magick-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/MagickCore-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/MagickCore-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/MagickWand-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/MagickWand-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config -config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Wand-config'
substituteStream(): WARNING: pattern '/nix/store/pjrkzmdm3p93r2la8j5bvksh5cq84535-pkg-config-wrapper-0.29.2/bin/pkg-config' doesn't match anything in file '/nix/store/74pfrls4dp38330bvxkgaiqapn17mqqx-imagemagick-6.9.11-57-dev/bin/Wand-config'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by dotlambda** [comment]
closing in favour of https://github.com/NixOS/nixpkgs/pull/111287

**Pull request closed without merging**